pallets/corporate-actions/src/lib.rs

```diff
@@ -767,6 +725,67 @@ decl_module! {
             CorporateActions::insert(ca_id.ticker, ca_id.local_id, ca.clone());
             Self::deposit_event(Event::RecordDateChanged(agent, ca_id, ca));
         }
+
+         #[weight = <T as Config>::WeightInfo::initiate_corporate_action_use_defaults(
+                T::MaxDidWhts::get(),
+                T::MaxTargetIds::get(),
+            )
+            .max(<T as Config>::WeightInfo::initiate_corporate_action_provided(
+                ca_args.withholding_tax.as_ref().map_or(0, |whts| whts.len() as u32),
+                ca_args.targets.as_ref().map_or(0, |t| t.identities.len() as u32),
+            )).saturating_add(<T as Config>::DistWeightInfo::distribute())
+        ]
+        pub fn initiate_corporate_action_and_distribute(
+            origin,
+            ca_args: InitiateCorporateActionArgs,
+            portfolio: Option<PortfolioNumber>,
+            currency: Ticker,
+            per_share: Balance,
+            amount: Balance,
+            payment_at: Moment,
+            expires_at: Option<Moment>,
+        ) -> DispatchResult {
+            let InitiateCorporateActionArgs {
+                ticker,
+                kind,
+                decl_date,
+                record_date,
+                details,
+                targets,
+                default_withholding_tax,
+                withholding_tax
+            } = ca_args;
+
+            let PermissionedCallOriginData {
+                primary_did: agent,
+                secondary_key,
+                ..
+            } = <ExternalAgents<T>>::ensure_agent_asset_perms(origin, ticker)?;
+
+            let ca_id = Self::unsafe_initiate_corporate_action(
+                agent.for_event(),
+                ticker,
+                kind,
+                decl_date,
+                record_date,
+                details,
+                targets,
+                default_withholding_tax,
+                withholding_tax
+            )?;
+
+            <distribution::Module<T>>::unsafe_distribute(
+                agent,
+                secondary_key,
+                ca_id,
+                portfolio,
+                currency,
+                per_share,
+                amount,
+                payment_at,
+                expires_at,
+            )
```

Comment:
Is the lack of `with_transaction` intentional? (if so, leave a comment in the code to that effect)

Reply:
Good catch. We should discuss.

Reply:
We decided in the meeting to consult the preference of SDK team, cc @monitz87